### ISSUE TITLE
Update Master record with good data if it was overwritten with blank data

### ISSUE THREAD

**Issue by RDmitchell** [SEED-platform/seed#611]
Per #610, we need to figure out a way to correct the problem in an existing organization's data. The program has a record of all the merged data, so the original data that was overwritten by blank data is known. 

We will work on this the first week of Jan when we are all back from the holidays. I am assigning it for now to @StephenCzarnecki

**Comment by RDmitchell** [comment]
@pipermerriam -- I am assigning this to you as your first priority for the next round of issues. This might be a bit tricky. If you can at least ponder it, and figure out how hard this would be to do, that would be a start. Thx !!

**Comment by RDmitchell** [comment]
One question about this is whether this is something we do "behind the scenes" on individual orgs that request it, or we put the functionality into the UI so a user can trigger this action as needed. If we decide to put it in the UI, it could be a superuser function, or it could be something that individual users (probably "owners") of an organization, could trigger. We can figure this part out once we figure out how hard it will be to do.

**Comment by pipermerriam** [comment]
@RDmitchell I think I'd prefer to do this as a data migration.  That way it's a one-off thing that runs on the next deploy and we don't end up with code that needs to be maintained long term since the application should now treat this data correctly.  Thoughts?

**Comment by RDmitchell** [comment]
That does seem like the right approach.

**Comment by RDmitchell** [comment]
I am assigning this to Stephen for testing.

**Comment by StephenCzarnecki** [comment]
In short looks good to me.  

I don't know how to write an automated test to test a migration so I just did it by hand.  I made some files with the following columns: tax lot id, address line 1, foo.  For some of the later files to prevent duplicates I added a "q" column too.  The idea being that neither "foo" nor "q" should be recognized and therefore would be extra data.  

I came up with a handful (18) of test cases some of which I now realize are probably redundant.

I started on revision 04cb03d0ff871ba6bf73e9ff18af027796e6e9ef but had problems viewing the building details so I moved to c6eccd95d3d6ada6e6334a1f6d68ff75991931e6 (commit Bump version number to 1.3.0)

I uploaded file extra_data_file_1.csv in the attached.  On the mapping screen I unchecked the MAP checkbox for the "foo" column.  Found no matches as expected.

Then I uploaded extra_data_file_2.csv.  The MAP checkbox next to the foo column was already uncheacked.  Matching found 19 as expected.

I verified that I was seeing the issue on the cases I expected to see it (1,2,6,7,8,12,13) and the other cases behaved as expected.

Then I uploaded the rest of the files in order clearing the MAP box next to the "q" column when necessary.  The overwriting behavior was present in all expected cases.

Then I just grabbed Piper's merge https://github.com/pipermerriam/seed/raw/db23637266ae1ede44007c36015ce86be4dbf9c1/seed/migrations/0012_auto_20151222_1031.py and applied that.

Then I verified that all the records had the expected value for the "foo" column.  That is, the Master record had the furthest-left non-blank "foo" value

Here is one example of a record that originally had one value in the "foo" field ("bar") then that was replaced with a new value ("baz") and finally had a file with a blank value for "foo".  Now the master record correctly show's the latest value ("baz")

![screenshot from 2016-01-05 01 27 31](https://cloud.githubusercontent.com/assets/7605407/12105341/c6410332-b34b-11e5-933c-9dae05c8b666.png)



[extra_data_issue.zip](https://github.com/SEED-platform/seed/files/77907/extra_data_issue.zip)

**Comment by StephenCzarnecki** [comment]
I ran the migration on the dump of the production database provided by Miles (seedprod-pg94-20151223).  It completed in around 38 minutes with the caveat that it needed a bit more memory than I had initially provided.  I got distracted by having my picture taken so don't have exact figures but it needed at least an additional 8GB of swap in addition to the 2GB of memory the machine also had free.  I actually ran it with 12GB of swap at the end.

More details:

All were run on my Ubuntu VM:
$ uname -a
Linux vagrant-ubuntu-trusty-64 3.13.0-40-generic #69-Ubuntu SMP Thu Nov 13 17:53:56 UTC 2014 x86_64 x86_64 x86_64 GNU/Linux

The machine was initially configured to have 4GB of memory and no swap file.  Running the migration on this configuration caused a seg fault due to lack of memory within a couple minutes.

I then gave the machine 4GB of swap and that ran for about 15 minutes before seg faulting due to lack of memory.

Then I gave the machine a total of 12GB of swap and it completed in a little under 38 minutes.
